packages/vm/src/calls.rs

```diff
@@ -8,30 +10,43 @@ use cosmwasm_std::{
 use crate::errors::{RuntimeErr, VmResult, WasmerRuntimeErr};
 use crate::instance::{Func, Instance};
 use crate::serde::{from_slice, to_vec};
+use schemars::JsonSchema;
 
 static MAX_LENGTH_INIT: usize = 100_000;
 static MAX_LENGTH_HANDLE: usize = 100_000;
 static MAX_LENGTH_QUERY: usize = 100_000;
 
-pub fn call_init<S: Storage + 'static, A: Api + 'static, Q: Querier + 'static>(
+pub fn call_init<S, A, Q, U>(
     instance: &mut Instance<S, A, Q>,
     env: &Env,
     msg: &[u8],
-) -> VmResult<Result<InitResponse, ApiError>> {
+) -> VmResult<Result<InitResponse<U>, ApiError>>
+where
+    S: Storage + 'static,
+    A: Api + 'static,
+    Q: Querier + 'static,
+    U: DeserializeOwned + Clone + fmt::Debug + JsonSchema + PartialEq,
+{
     let env = to_vec(env)?;
     let data = call_init_raw(instance, &env, msg)?;
-    let res: InitResult = from_slice(&data)?;
+    let res: InitResult<U> = from_slice(&data)?;
     Ok(res.into())
 }
 
-pub fn call_handle<S: Storage + 'static, A: Api + 'static, Q: Querier + 'static>(
+pub fn call_handle<S, A, Q, U>(
     instance: &mut Instance<S, A, Q>,
     env: &Env,
     msg: &[u8],
-) -> VmResult<Result<HandleResponse, ApiError>> {
+) -> VmResult<Result<HandleResponse<U>, ApiError>>
+where
+    S: Storage + 'static,
+    A: Api + 'static,
+    Q: Querier + 'static,
+    U: DeserializeOwned + Clone + fmt::Debug + JsonSchema + PartialEq,
```

Comment:
Can `U` get the default value as well, to avoid all the `call_handle::<_, _, _, NoMsg>` diffs here?

Reply:
I understand your desire, I did try that already, but got:

```
Defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions
```

you cannot use those defaults on a function call. That said, little contract code needs to use `call_handle`, only when they expect Wasm runtime error (catching panics, out of gas, etc)

Reply:
2x good point, thanks